qualtran/bloqs/block_encoding/sparse_matrix.py

```diff
@@ -0,0 +1,279 @@
+#  Copyright 2024 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import abc
+from functools import cached_property
+from typing import cast, Dict, Tuple
+
+import numpy as np
+from attrs import frozen
+
+from qualtran import (
+    Bloq,
+    bloq_example,
+    BloqBuilder,
+    BloqDocSpec,
+    BoundedQUInt,
+    QAny,
+    QBit,
+    QUInt,
+    Register,
+    Signature,
+    Soquet,
+    SoquetT,
+)
+from qualtran.bloqs.basic_gates import Rx, Swap
+from qualtran.bloqs.block_encoding import BlockEncoding
+from qualtran.bloqs.block_encoding.lcu_select_and_prepare import PrepareOracle
+from qualtran.bloqs.state_preparation.prepare_uniform_superposition import (
+    PrepareUniformSuperposition,
+)
+from qualtran.symbolics import SymbolicFloat, SymbolicInt
+
+
+@frozen
+class RowColumnOracle(Bloq):
```

Comment:
Granted I'm not an expert on exactly why, but If this is supposed to be an abstract class -- which I think it is since it has abstract methods -- I think you're supposed to add `(Bloq, metaclass=abc.ABCMeta)` to the inheritance

Reply:
Actually, if we want to really get precise, inheriting from `Bloq` was sufficient here. A child of a class with `metaclass=abc.ABCMeta` gets the same benefit as its parent.

```
>>> class A(metaclass=abc.ABCMeta):
...     pass
... 
>>> class B(A):
...     @abc.abstractmethod
...     def f(): ...
... 
>>> B()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: Can't instantiate abstract class B with abstract method f
```

`Bloq` already declares the `metaclass` here. I guess inheriting from `abc.ABC` makes it clear how this benefit would pass down to child classes.